mne/io/persyst/persyst.py

```diff
@@ -0,0 +1,407 @@
+        testdate = patient_dict.get('testdate')
+        if testdate is not None:
+            # TODO: Persyst may change its internal date schemas
+            #  without notice
+            # These are the 3 "so far" possible datatime storage
+            # formats in Persyst .lay
+            if '/' in testdate:
+                time.strptime()
+                testdate = datetime.strptime(testdate, '%m/%d/%Y')
+            elif '-' in testdate:
+                testdate = datetime.strptime(testdate, '%d-%m-%Y')
+            elif '.' in testdate:
+                testdate = datetime.strptime(testdate, '%Y.%m.%d')
+
+            if not isinstance(testdate, datetime):
+                warn('Cannot read in the measurement date due '
+                     'to incompatible format. Please set manually '
+                     'for %s ' % lay_fname)
+            else:
+                testtime = datetime.strptime(patient_dict.get('testtime'),
+                                             '%H:%M:%S')
+                meas_date = datetime(
+                    year=testdate.year, month=testdate.month,
+                    day=testdate.day, hour=testtime.hour,
+                    minute=testtime.minute, second=testtime.second,
+                    tzinfo=timezone.utc)
+
+        # Create mne structure
+        ch_names = list(channelmap_dict.keys())
+        if n_chs != len(ch_names):
+            raise RuntimeError('Channels in lay file do not '
+                               'match the number of channels '
+                               'in the .dat file.')  # noqa
+        # get rid of the "-Ref" in channel names
+        ch_names = [ch.split('-ref')[0] for ch in ch_names]
+
+        # get the sampling rate and default channel types to EEG
+        sfreq = fileinfo_dict.get('samplingrate')
+        ch_types = 'eeg'
+        info = create_info(ch_names, sfreq, ch_types=ch_types)
+        info.update(subject_info=subject_info)
+        for idx in range(n_chs):
+            # calibration brings to uV then 1e-6 brings to V
+            info['chs'][idx]['cal'] = cal * 1.0e-6
+        info['meas_date'] = meas_date
+
+        # determine number of samples in file
+        # Note: We do not use the lay file to do this
+        # because clips in time may be generated by Persyst that
+        # DO NOT modify the "SampleTimes" section
+        with open(dat_fpath, 'rb') as f:
+            # determine the precision
+            if int(datatype) == 7:
+                # 32 bit
+                dtype = np.dtype('i4')
+            elif int(datatype) == 0:
+                # 16 bit
+                dtype = np.dtype('i2')
+            else:
+                raise RuntimeError(f'Unknown format: {datatype}')
+
+            # allow offset to occur
+            f.seek(0, os.SEEK_END)
+            n_samples = f.tell()
+            n_samples = n_samples // (dtype.itemsize * n_chs)
+
+            if verbose:
+                logger.debug(f'Loaded {n_samples} samples '
```

Comment:
No need to nest in `if verbose`, it magically gets accounted for
```suggestion
            logger.debug(f'Loaded {n_samples} samples '
```